plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,177 +82,330 @@ function executeCLI($scriptName, &$args) {
-						$submission->setUserId($user->getId());
-						$submission->stampLastActivity();
-						$submission->setStatus(STATUS_PUBLISHED);
-						$submission->setWorkType($isEditedVolume == 1?WORK_TYPE_EDITED_VOLUME:WORK_TYPE_AUTHORED_WORK);
-						$submission->setCopyrightNotice($press->getLocalizedSetting('copyrightNotice'), $locale);
-						$submission->setLocale($locale);
-
-						$series = $seriesDao->getByPath($seriesPath, $press->getId());
-						if ($series) {
-							$submission->setSeriesId($series->getId());
+				// we need a Genre for the files.  Assume a key of MANUSCRIPT as a default.
+				$genre = $genreDao->getByKey('MANUSCRIPT', $press->getId());
+				if (!$genre) {
+					echo __('plugins.importexport.csv.noGenre') . "\n";
+					exit();
+				}
+
+				$authorGroup = $userGroupDao->getDefaultByRoleId($press->getId(), ROLE_ID_AUTHOR);
+
+				if (!$authorGroup) {
+					echo __('plugins.importexport.csv.noAuthorGroup', ['press' => $pressPath]) . "\n";
+					continue;
+				}
+
+				$submission = $submissionDao->newDataObject();
+				$submission->setContextId($press->getId());
+				$submission->stampLastActivity();
+				$submission->stampModified();
+				$submission->setStatus(STATUS_PUBLISHED);
+				$submission->setWorkType($isEditedVolume == 1 ? WORK_TYPE_EDITED_VOLUME : WORK_TYPE_AUTHORED_WORK);
+				$submission->setCopyrightNotice($press->getLocalizedSetting('copyrightNotice'), $locale);
+				$submission->setLocale($locale);
+				$submission->setStageId(WORKFLOW_STAGE_ID_PRODUCTION);
+				$submission->setSubmissionProgress(0);
+				$submission->setData('abstract', $abstract, $locale);
+
+				$series = $seriesPath ? $seriesDao->getByPath($seriesPath, $press->getId()) : null;
+				if ($series) {
+					$submission->setSeriesId($series->getId());
+				}
+
+				$submissionId = $submissionDao->insertObject($submission);
+
+				$publication = $publicationDao->newDataObject();
+				$publication->setData('submissionId', $submissionId);
+				$publication->setData('version', 1);
+				$publication->setData('status', STATUS_PUBLISHED);
+				$publication->setData('datePublished', Core::getCurrentDate());
+				$publicationId = $publicationDao->insertObject($publication);
+
+				$submission->setData('currentPublicationId', $publicationId);
+				$submissionDao->updateObject($submission);
+
+				$contactEmail = $press->getContactEmail();
+				$authorsString = trim($authorString, '"'); // remove double quotes if present.
+				$authors = preg_split('/;/', $authorsString);
+				$firstAuthor = true;
+
+				foreach ($authors as $authorString) {
+					// Examine the author string. Best case is: Given1 Family1 <email@address.com>, Given2 Family2 <email@address.com>, etc
+					// But default to press email address based on press path if not present.
+					$givenName = $familyName = $emailAddress = null;
+					$authorString = trim($authorString); // whitespace.
+					if (!preg_match('/^([\w.\s]+)\s+([\w\s-]+)?\s*(<([^>]+)>)?$/', $authorString, $matches)) {
```

Comment:
Should this be `if (preg_match...` rather than a "not"? My test import with the sample data is throwing an error because the email address isn't set, but it worked once I changed this.